includes/class-newspack-popups-model.php

```diff
@@ -993,6 +988,11 @@ function( $acc, $segment_id ) {
 		}
 
 		foreach ( $analytics_events as &$event ) {
+			// If a form submission and the form contains registration + list info, append that to the event label.
+			if ( 'submit' === $event['on'] && $has_register_form ) {
+				$event_label .= __( ' | ${formId} - ${formFields[lists[]]}' );
```

Comment:
I've seen two issues: 

1. This string is attached also to the form dismissal event label, which is unnecessary
2. The above happens twice, somehow. 

Here are the last two items on `$analytics_events` in my testing:

```
    [3] => Array
        (
            [amp_on] => amp-form-submit
            [on] => submit
            [element] => #c529 form:not(.c529)
            [event_name] => Form Submission
            [id] => c27c
            [event_category] => Newspack Announcement
            [event_label] => Center: Register (343) - Anons | ${formId} - ${formFields[lists[]]}
        )

    [4] => Array
        (
            [amp_on] => amp-form-submit-success
            [on] => submit
            [element] => .c5290
            [event_name] => Dismissal
            [non_interaction] => 1
            [id] => cda1
            [event_category] => Newspack Announcement
            [event_label] => Center: Register (343) - Anons | ${formId} - ${formFields[lists[]]} | ${formId} - ${formFields[lists[]]}
        )
```

Reply:
@aschweigert are these two events considered redundant? If so, which one should we keep for dismissal? We use the `Form Submission` one for non-dismissal form submissions.

Reply:
They don't appear to be redundant...we definitely wouldn't want to consider a dismissal a successful form submission, is that what's happening currently?

Reply:
Correct, a prompt dismissal is technically considered a form submission. So it looks like we're capturing that erroneously.

Reply:
Yep, that's very much the opposite of a successful form submission :) so we should definitely track these as separate events

Reply:
Both issues should be fixed by 1e1da62.

Reply:
> Correct, a prompt dismissal is technically considered a form submission. So it looks like we're capturing that erroneously.

But these are two different events. On dismissal, a `Dismissal` event is sent, and on form submission a `Form Submission` event is sent. So all was well, the change in 1e1da62fbc920b77c7c1bd2862b4455c56627423 should not be needed. Or am I missing something?

Reply:
Also, the first issue from my original comment is still reproducible – the `Dismissal` event still get the `| ${formId} - ${formFields[lists[]]}` appended. This is probably because `$event_label` is declared outside of the loop, so on iterations after it's first modified, it will still be modified.

Reply:
> But these are two different events. On dismissal, a `Dismissal` event is sent, and on form submission a `Form Submission` event is sent. So all was well, the change in 1e1da62fbc920b77c7c1bd2862b4455c56627423 should not be needed. Or am I missing something?

@aschweigert what are your thoughts on this? It sounded to me like you were not expecting dismissals to also result in a form submission event. What is desired from an analytics standpoint?

Reply:
@dkoo Sorry if that wasn't clear! The dismissal event should *not* be a form submission event, it should be tracked separately.

Reply:
That is currently the case, but what @adekbadek explains above is that a dismissal is technically also a form submission, so it results in two events: dismissal, and form submission. Should the form submission event not happen at all in this case?

Reply:
ah gotcha, yeah we should definitely not fire the form submission event on the dismissal, that should make the analysis a bit easier

Reply:
@adekbadek 363829b fixes the definition of `$event_label` so that the extra form info isn't appended to non-form-submission events.